back/engines/free/surveys/app/services/surveys/typeform_api_parser.rb

```diff
@@ -41,8 +41,14 @@ def get_responses(form_id)
 
     def extract_field_titles(tf_form)
       tf_form['fields'].map do |f|
-        [f['id'], f['title']]
-      end.to_h
+        if f['properties'] && f['properties']['fields']
+          f['properties']['fields'].map do |subf|
+            [subf['id'], subf['title']]
+          end
+        else
+          [[f['id'], f['title']]]
+        end
+      end.flatten(1).to_h
```

Comment:
Could something like that work? (didn't test it)
```ruby
def extract_field_titles(tf_form)
  tf_form['fields'].flat_map do |top_field|
    fields = f.dig('properties', 'fields') || [top_field]
    fields.map {|f| f.pluck('id', 'title') }
  end.to_h
end
```

Could be a nice case for recursion too 😵‍💫